File: ValidAnagram_242/solution.ts
function isAnagram(s: string, t: string): boolean {
  if (s.length !== t.length) return false;

  const hashS = new Map<string, number>();
  const hashT = new Map<string, number>();

  for (let i = 0; i < s.length; i++) {
    const countS = hashS.get(s[i]) || 0;
    hashS.set(s[i], countS + 1);

    const countT = hashT.get(t[i]) || 0;
    hashT.set(t[i], countT + 1);
  }

  for (const key of hashS.keys()) {
    if (hashS.get(key) !== hashT.get(key)) return false;
  }

  return true;
}

function isAnagram2(s: string, t: string): boolean {
  if (s.length !== t.length) return false;

  const sortedS = [...s].sort().join("");
  const sortedT = [...t].sort().join("");

  return sortedS === sortedT;
}

File: ValidAnagram_242/solution.cs
public class Solution {
    public bool IsAnagram(string s, string t) {
        if(s.Length != t.Length) return false;

        var hashS = new Dictionary<char, int>();
        var hashT = new Dictionary<char, int>();

        for (int i = 0; i < s.Length; i++)
        {
            hashS[s[i]] = hashS.ContainsKey(s[i]) ? hashS[s[i]]++ : 1;
            hashT[t[i]] = hashT.ContainsKey(t[i]) ? hashT[t[i]]++ : 1;
        }

        foreach (var item in hashS)
        {
            if(!hashT.ContainsKey(item.Key) || hashT[item.Key] != item.Value) return false;
        }

        return true;
    }

    public bool IsAnagram2(string s, string t) {
        if(s.Length != t.Length) return false;

        var sortedS = new string(s.OrderBy(c => c).ToArray());
        var sortedT = new string(t.OrderBy(c => c).ToArray());

        return sortedS == sortedT;
    }
}
